prov/efa/test/efa_unit_test_info.c

```diff
@@ -151,3 +152,90 @@ void test_info_check_shm_info()
 	test_info_check_shm_info_from_hints(hints);
 
 }
+
+/**
+ * @brief Check that case when a user requested FI_HMEM support
+ *        using libfabric API < 1.18,
+ */
+void test_info_check_hmem_cuda_support_on_api_lt_1_18()
+{
+	struct fi_info *hints, *info = NULL;
+	int err;
+
+	if (!hmem_ops[FI_HMEM_CUDA].initialized)
+		skip();
+
+	hints = efa_unit_test_alloc_hints(FI_EP_RDM);
+
+	hints->caps |= FI_HMEM;
+	hints->domain_attr->mr_mode |= FI_MR_HMEM;
+
+	/* For libfabric API < 1.18,
+	 * on a system that support GPUDirect RDMA read,
+	 * HMEM cuda is available when GPUDirect RDMA is available,
+	 * and environment variable FI_EFA_USE_DEVICE_RDMA set to 1/on/true
+	 * otherwise it is not available.
+	 */
+	setenv("FI_EFA_USE_DEVICE_RDMA", "1", true /* overwrite */);
+	err = fi_getinfo(FI_VERSION(1, 6), NULL, NULL, 0, hints, &info);
+	if (efa_device_support_rdma_read()) {
+		assert_int_equal(err, 0);
+		fi_freeinfo(info);
```

Comment:
Being pedantic - `fi_freeinfo` should be called before assertion. Same in `test_info_check_hmem_cuda_support_on_api_ge_1_18`

Reply:
I do not think so. If err is not 0, we should not freeinfo.

Reply:
I see. You are right. I was mistaken.